server/test/unit/paTest/handleCheckDatabaseConnectionTest.js

```diff
@@ -0,0 +1,74 @@
+// Third-party dependencies
+const { expect, use } = require('chai')
+const { describe, it } = require('mocha')
+const sinon = require('sinon')
+const sinonChai = require('sinon-chai')
+
+// In-house dependencies
+const { helpers } = require('brave-alert-lib')
+const { mockResponse } = require('../../../testingHelpers')
+const pa = require('../../../pa')
+const db = require('../../../db/db')
+
+const testGoogleIdToken = 'google-id-token'
+const braveKey = helpers.getEnvVar('PA_API_KEY_PRIMARY')
+
+use(sinonChai)
+const sandbox = sinon.createSandbox()
+
+describe('pa.js unit tests: handleCheckDatabaseConnection', () => {
+  beforeEach(() => {
+    this.res = mockResponse(sandbox)
+    sandbox.spy(helpers, 'logError')
+  })
+
+  afterEach(() => {
+    sandbox.restore()
+  })
+
+  describe('for a valid request where the database function successfully returns a client', () => {
+    beforeEach(async () => {
+      await pa.handleCheckDatabaseConnection(
+        {
+          body: { braveKey, googleIdToken: testGoogleIdToken },
+        },
+        this.res,
+      )
+    })
+    it('should respond with status 200', () => {
+      expect(this.res.status).to.be.calledWith(200)
+    })
+  })
+
+  describe('for an request with an invalid braveKey', () => {
+    beforeEach(async () => {
+      await pa.handleCheckDatabaseConnection(
+        {
+          body: { braveKey: 'invalidKey', googleIdToken: testGoogleIdToken },
+        },
+        this.res,
+      )
+    })
+    it('should respond with status 401', () => {
+      expect(this.res.status).to.be.calledWith(401)
+    })
+  })
+
+  describe('for a valid request where an error occurs during database access', () => {
+    beforeEach(async () => {
+      sandbox.stub(db, 'getCurrentTime').rejects(new Error('fake error'))
```

Comment:
This looks good.

Can you please also stub the `db.getCurrentTime` call for the other tests too? Because these are unit tests, we don't want them actually accessing the DB otherwise they could unexpectedly fail. For the other tests, you'll set it up so that the DB call returns successfully.

Reply:
Yes will do!